compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -379,7 +379,9 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
                 // avoid type aliases for tuples
                 Right(MirrorSource.GenericTuple(types))
               case _ => reduce(tp.underlying)
-          case tp: MatchType => reduce(tp.tryNormalize.orElse(tp.superType))
+          case tp: MatchType =>
+            val n = tp.tryNormalize
+            if n.exists then reduce(n) else Left(i"its subpart `$tp` is an unreducible match type.")
```

Comment:
I think we should return the unreduced match type instead.
Perhaps an instead was provided manually for it.

Reply:
`reduce` requires that we return a class symbol it seems. Also the Synthesizer logic should only be used as a fallback if an implicit wasn't found by regular search.

Reply:
Ah good point, then an error sounds fine to me.